File: Data Structures & Algorithms/binary-tree-diameter/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int diameter = -1;
    private int helper(TreeNode root) {
        if (root == null) return 0;

        int left = helper(root.left);
        int right = helper(root.right);

        diameter = Math.max(diameter, left + right);
        return Math.max(0, Math.max(left, right)) + 1;
    }
    public int diameterOfBinaryTree(TreeNode root) {
        if (root == null) return 0;
        helper(root);
        return diameter;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def helper(self, root: TreeNode, max_val: int) -> int:
        if root is None:
            return 0
        
        ans = 1 if root.val >= max_val else 0

        left_ans = self.helper(root.left, max(root.val, max_val))
        right_ans = self.helper(root.right, max(root.val, max_val))

        return ans + left_ans + right_ans
    
    def goodNodes(self, root: TreeNode) -> int:
        if root is None:
            return 0

        return self.helper(root, root.val)
        